osu.Game/Overlays/RankingsOverlay.cs

```diff
@@ -90,154 +78,112 @@ private void load()
+            var display = selectDisplay().With(d =>
+            {
+                d.Current = ruleset;
+                d.StartLoading = loading.Show;
+                d.FinishLoading = loading.Hide;
+            });
 
-            if (Scope.Value == RankingsScope.Spotlights)
+            if (display is SpotlightsDisplay)
             {
-                loadContent(new SpotlightsLayout
-                {
-                    Ruleset = { BindTarget = ruleset }
-                });
+                loadSpotlightsDisplay((SpotlightsDisplay)display);
                 return;
             }
 
-            var request = createScopedRequest();
-            lastRequest = request;
+            loadDisplayAsync(display);
+        }
 
-            if (request == null)
+        private void loadSpotlightsDisplay(SpotlightsDisplay display)
+        {
+            spotlightsRequest = new GetSpotlightsRequest();
+            spotlightsRequest.Success += response => Schedule(() =>
             {
-                loadContent(null);
-                return;
-            }
-
-            request.Success += () => Schedule(() => loadContent(createTableFromResponse(request)));
-            request.Failure += _ => Schedule(() => loadContent(null));
+                display.Spotlights = response.Spotlights;
+                loadDisplayAsync(display);
+            });
+            API.Queue(spotlightsRequest);
+        }
 
-            api.Queue(request);
+        private void loadDisplayAsync(RankingsDisplay display)
+        {
+            LoadComponentAsync(display, loaded =>
+            {
+                contentContainer.Child = loaded;
+            }, (cancellationToken = new CancellationTokenSource()).Token);
         }
 
-        private APIRequest createScopedRequest()
+        private RankingsDisplay selectDisplay()
         {
             switch (Scope.Value)
             {
-                case RankingsScope.Performance:
-                    return new GetUserRankingsRequest(ruleset.Value, country: Country.Value?.FlagName);
-
                 case RankingsScope.Country:
-                    return new GetCountryRankingsRequest(ruleset.Value);
+                    return new CountriesDisplay();
+
+                case RankingsScope.Performance:
+                    return new PerformanceDisplay
+                    {
+                        Country = { BindTarget = Country }
+                    };
 
                 case RankingsScope.Score:
-                    return new GetUserRankingsRequest(ruleset.Value, UserRankingsType.Score);
+                    return new ScoresDisplay();
+
+                case RankingsScope.Spotlights:
+                    return new SpotlightsDisplay();
             }
 
-            return null;
+            throw new NotImplementedException($"Display for {Scope.Value} is not implemented.");
         }
 
-        private Drawable createTableFromResponse(APIRequest request)
+        public void ShowCountry(Country requested)
```

Comment:
the flag clicks use this method anyway, so why not get rid of the `Country` bindable entirely?

Reply:
Well, we don't want to reload the whole display by changing the country. If current display is `Performance` - we just need to pass the `Country` value so it will reload it's content. Probably we can check if selected overlay is performance one, but it will be worse imo

Reply:
you should be able to move out three logic in the `BindValueChanged` callback to this method, no? at which point the bindable has no more purpose.

Reply:
Right, but there's another purpose of this bindable: to pass `Country` value to the `PerformanceDisplay`